data-acquisition/data-synthesizer/src/data_synthesizer/data_synthesizer.py

```diff
@@ -0,0 +1,214 @@
+        curr_obj_conf = self.fg_conf[curr_fg_category][curr_fg_obj]
+        fg_class_id = self.class_map[curr_fg_category]
+
+        curr_obj_file = (
+            self.template_path + "/foregrounds/{cat}/{obj}.png").format(
+                cat=curr_fg_category, obj=curr_fg_obj)
+
+        return curr_obj_file, curr_obj_conf, fg_class_id
+
+    def _get_bg_and_conf(self):
+        curr_bg_label = np.random.choice(self.bg_labels, 1)[0]
+        curr_bg_conf = self.bg_conf[curr_bg_label]
+        curr_bg_file = (
+            self.template_path + "/backgrounds/{bg}.jpg").format(
+                bg=curr_bg_label)
+        return curr_bg_file, curr_bg_conf
+
+    def _process_shelf(
+            self, shelf, bg_conf, bg_file, image_path, categories, rot_pc,
+            obj_sizes_allowed, max_objs_in_pack, max_offset):
+        """
+        Target function for parallely processing each individual shelf. 
+        """
+        shelf_region = bg_conf["shelf_region"]
+        x_start, x_end = shelf_region[0][0], shelf_region[1][0]
+        shelf_positions = bg_conf["shelf_y_positions"]
+        shelf_ht = bg_conf["shelf_ht"]
+
+        bg_img = Image.open(bg_file)
+        empty_fg = Image.new('RGBA', bg_img.size, color=(0, 0, 0, 0))
+        empty_alpha_mask = Image.new('L', bg_img.size, color=0)
+        shelf_alpha_mask = empty_alpha_mask.copy()
+        num_obj = 0
+
+        paste_pos = (x_start, 0)
+        while paste_pos[0]  < x_end:
+            fg_file, fg_conf, fg_id = self._get_fg_and_conf(categories)
+            new_size = self._get_new_image_size(
+                fg_conf, obj_sizes_allowed, shelf_ht)
+            fg_img = Image.open(fg_file)
+            fg_img = fg_img.resize(new_size, Image.LANCZOS)
+
+            num_instances = np.random.choice(max_objs_in_pack, 1)[0]
+
+            for _ in range(num_instances):
+                num_obj += 1
+                is_rotated = np.random.binomial(1, p=rot_pc)
+                to_paste = fg_img
+                if is_rotated:
+                    to_paste = fg_img.rotate(
+                        np.random.randint(-90, 90), expand=1)
+
+                img_size = to_paste.size
+                alpha_mask = to_paste.getchannel(3)
+                paste_pos = (paste_pos[0],
+                             shelf_positions[shelf]-img_size[1])
+                
+                if paste_pos[0] + img_size[0] > x_end:
+                    paste_pos = (paste_pos[0]+img_size[0],
+                        shelf_positions[shelf])
+                    break
+
+                new_fg = empty_fg.copy()
+                new_fg.paste(to_paste, paste_pos)
+
+                new_alpha_mask = empty_alpha_mask.copy()
+                new_alpha_mask.paste(alpha_mask, paste_pos)
+
+                shelf_alpha_mask.paste(alpha_mask, paste_pos)
+
+                new_alpha_mask.save(
+                    image_path + "/train_mask/mask_{}{}${}.png".format(
```

Comment:
Can you change the delimiter from '$' to '___'(triple underscore) as this is what is used in the pre-process script.